linkis-computation-governance/linkis-engineconn/linkis-engineconn-executor/accessible-executor/src/main/scala/org/apache/linkis/engineconn/acessible/executor/service/DefaultAccessibleService.scala

```diff
@@ -92,15 +92,12 @@ class DefaultAccessibleService extends AccessibleService with Logging {
       return
     }
     var executor: Executor = ExecutorManager.getInstance.getReportExecutor
-    if (null != executor) {
-      Utils.tryAndWarn {
-        executor.close()
-        executor.tryShutdown()
-      }
-      logger.warn(s"Engine : ${Sender.getThisInstance} with state has stopped successfully.")
-    } else {
+    if (null == executor) {
       executor = SensibleExecutor.getDefaultErrorSensibleExecutor
     }
+    Utils.tryAndWarn(executor.tryShutdown())
```

Comment:
Thank you, is it because ReportExecutor is null?

Reply:
No, the root cause is because first close, then tryShutdown. And It is logical to change the state (such as tryShutdown, tryFailed) first and then release the resource(such as close)
OnceExecutor will be affected:
![image](https://user-images.githubusercontent.com/29919212/221729404-52da06b7-eba5-4544-a202-e79da0344dab.png)

not OnceExecutor will not be affected:
![image](https://user-images.githubusercontent.com/29919212/221730031-d753baf2-3919-46fc-81ea-fa5c1bb2e850.png)

Reply:
Get